test/com/amazon/ion/impl/TextWriterTest.java

```diff
@@ -487,6 +487,25 @@ public void testWritingJsonLongClobs()
         assertEquals("\"a\\\"'\\nc\\u007f\"", actual);
     }
 
+    @Test
+    public void testWritingJsonNanAndInfinity()
+            throws Exception
+    {
+        options = IonTextWriterBuilder.json();
+        options.setInitialIvmHandling(SUPPRESS);
+
+        IonDatagram dg = system().newDatagram();
+        dg.add().newFloat(Double.NaN);
+        dg.add().newFloat(Double.POSITIVE_INFINITY);
+        dg.add().newFloat(Double.NEGATIVE_INFINITY);
+
+        iw = makeWriter();
+        dg.writeTo(iw);
+
+        String actual = outputString();
+        assertEquals("null null null", actual);
+    }
```

Comment:
*Minor*: I think these should be broken out into separate tests IMO, but I am not going to block approval on it.